src/main/java/Chatbot/HarddiskStorage.java

```diff
@@ -0,0 +1,105 @@
+package Chatbot;
+
+import Tasks.Deadline;
+import Tasks.Event;
+import Tasks.Task;
+import Tasks.ToDo;
+
+import java.io.*;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+import java.time.LocalDateTime;
+
+/**
+ * The {@code HarddiskStorage} class handles the loading and saving of tasks to and from the hard disk.
+ * It supports various types of tasks, including ToDo, Deadline, and Event tasks.
+ */
+public class HarddiskStorage {
+    private final String filePath;
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm");
+
+    /**
+     * Constructs a {@code HarddiskStorage} object with the specified file path.
+     *
+     * @param filePath the path of the file where tasks are stored.
+     */
+    public HarddiskStorage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads tasks from the file specified by the {@code filePath}.
+     * If the file does not exist or cannot be read, an empty task list is returned.
+     *
+     * @return a map of tasks, where the key is the task ID and the value is the task object.
+     */
+    public Map<Integer, Task> load() {
+        Map<Integer, Task> tasks = new HashMap<>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(" \\| ");
+                int taskId = Integer.parseInt(parts[0]);
```

Comment:
perhaps there is no need to store the `taskId` of each task in the save file, instead the id could be appended to the tasks as they are loaded in from the save file since the tasks will be loaded in order of the sequence they are in in the save file